warehouse/migrations/versions/f7720656a33c_index_ultranormalized_pending_project_.py

```diff
@@ -0,0 +1,58 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Index ultranormalized pending project_name
+
+Revision ID: f7720656a33c
+Revises: 26455e3712a2
+Create Date: 2024-08-20 06:07:46.546659
+"""
+
+from alembic import op
+
+revision = "f7720656a33c"
+down_revision = "26455e3712a2"
+
+# Note: It is VERY important to ensure that a migration does not lock for a
+#       long period of time and to ensure that each individual migration does
+#       not break compatibility with the *previous* version of the code base.
+#       This is because the migrations will be ran automatically as part of the
+#       deployment process, but while the previous version of the code is still
+#       up and running. Thus backwards incompatible changes must be broken up
+#       over multiple migrations inside of multiple pull requests in order to
+#       phase them in over multiple deploys.
+#
+#       By default, migrations cannot wait more than 4s on acquiring a lock
+#       and each individual statement cannot take more than 5s. This helps
+#       prevent situations where a slow migration takes the entire site down.
+#
+#       If you need to increase this timeout for a migration, you can do so
+#       by adding:
+#
+#           op.execute("SET statement_timeout = 5000")
+#           op.execute("SET lock_timeout = 4000")
+#
+#       To whatever values are reasonable for this migration as part of your
+#       migration.
+
+
+def upgrade():
+    op.execute(
+        """
+        CREATE INDEX pending_project_name_ultranormalized
+          ON pending_oidc_publishers (ultranormalize_name(project_name));
```

Comment:
I opted against `CREATE INDEX CONCURRENTLY` here because [the original migration didn't use it](https://github.com/pypi/warehouse/blob/8a5ad08a1bf65b769efaa97ad0f49b7739a20857/warehouse/migrations/versions/d18d443f89f0_ultranormalize_name_function.py#L45-L50) and I expect that there are orders of magnitude fewer `pending_oidc_publishers` rows than `projects` rows.